### PR TITLE
Lite: Slice Op refactored

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
Remove unnecessary reversal of input dimension

### CODE REVIEW

**Comment by ANSHUMAN87** [comment]
@jianlijianli : Gentle Reminder!

**Comment by ANSHUMAN87** [comment]
@jianlijianli : Gentle Reminder!

**Comment by ANSHUMAN87** [comment]
@jianlijianli : Gentle Reminder!

**Comment by ANSHUMAN87** [comment]
@jianlijianli : Gentle Reminder!

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by jianlijianli** [comment]
Could you please add a high level description what this CL does? Thanks.

**Comment by gbaned** [comment]
@ANSHUMAN87 Could you please address the reviewer comments and resolve conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by ANSHUMAN87** [comment]
> Could you please add a high level description what this CL does? Thanks.

Updated!

**Comment by ANSHUMAN87** [comment]
@jianlijianli: Gentle reminder!

**Review by jdduke — DISMISSED** [review]

**Comment by ANSHUMAN87** [comment]
@gbaned: Gentle reminder! Please help merge the PR!

File: tensorflow/lite/kernels/slice_test.cc [hunk 1]
```diff
@@ -238,6 +238,28 @@ TEST_P(SliceOpTest, SliceString) {
                                 "2,0,0,0", "2,0,1,0", "2,0,2,0"}));
 }
 
+TEST(SliceOpTest, SliceInt64) {
+  SliceOpModel<int64_t, int32_t> m({3, 2, 3, 1}, {4}, {4}, TensorType_INT32,
```
Comment (rthadur) [inline tensorflow/lite/kernels/slice_test.cc:242]: `tensorflow/lite/kernels/slice_test.cc:242:34: error: no matching constructor for initialization of 'SliceOpModel<int64_t, int32_t>' (aka 'SliceOpModel<long, int>')
  SliceOpModel<int64_t, int32_t>`

File: tensorflow/lite/kernels/slice_test.cc [hunk 2]
```diff
@@ -238,6 +238,28 @@ TEST_P(SliceOpTest, SliceString) {
...
+  m.SetInput({1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 5, 5, 6, 6, 6});
+  m.SetBegin({1, 0, 0, 0});
+  m.SetSize({2, 1, -1, 1});
+  m.Invoke();
+  EXPECT_THAT(m.GetOutputShape(), ElementsAreArray({2, 1, 3, 1}));
+  EXPECT_THAT(m.GetOutput(), ElementsAreArray({3, 3, 3, 5, 5, 5}));
+}
+
+TEST(SliceOpTest, SliceBool) {
+  SliceOpModel<bool, int32_t> m({2, 3}, {2}, {2}, TensorType_INT32,
```
Comment (rthadur) [inline tensorflow/lite/kernels/slice_test.cc:253]: `tensorflow/lite/kernels/slice_test.cc:253:31: error: no matching constructor for initialization of 'SliceOpModel<bool, int32_t>' (aka 'SliceOpModel<bool, int>')
  SliceOpModel<bool, int32_t> m({2, 3}, {2}, {2}, TensorType_INT32,`

**Review by rthadur — CHANGES_REQUESTED** [review]
@ANSHUMAN87 can you please check internal error ?

**Comment by gbaned** [comment]
@ANSHUMAN87 Can you please check @rthadur's comments and keep us posted. Thanks!

**Comment by ANSHUMAN87** [comment]
@gbaned , @rthadur : It is strange i am unable to reproduce the errors as you mentioned.

**Comment by ANSHUMAN87** [comment]
@gbaned , @rthadur : The failure is resolved now, Thanks!

**Comment by ANSHUMAN87** [comment]
@gbaned , @rthadur : I do not think there is anything left for this PR as @jdduke : already approved this PR earlier. Can you please expedite and merge this PR. Thanks!

**Comment by rthadur** [comment]
@ANSHUMAN87 sorry for the delay , will try to pull this again , thank you

**Review by rthadur — APPROVED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by ANSHUMAN87** [comment]
> @ANSHUMAN87 getting below error , can you please check
> `/tensorflow/lite/kernels/test_util.cc:317 Expected equality of these values: CountPartitionsDelegatedTo(interpreter_.get(), TestNnApiDelegate()) Which is: 0 1 Expecting operation to be accelerated but cannot find a partition associated to the NNAPI delegate`

@rthadur : Sorry for late response! The error you posted does not seem to be relevant to my change! Please let me know, if anything needs to be done from my end. Thanks!

**Comment by rthadur** [comment]
@jianlijianli @jdduke  can you please help with internal failures ?

File: tensorflow/lite/kernels/slice_test.cc [hunk 3]
```diff
@@ -238,6 +238,25 @@ TEST_P(SliceOpTest, SliceString) {
                                 "2,0,0,0", "2,0,1,0", "2,0,2,0"}));
 }
 
+TEST_P(SliceOpTest, SliceInt64) {
```
Comment (jdduke) [inline tensorflow/lite/kernels/slice_test.cc:241]: You'll need to blacklist these 2 new tests @ https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/nnapi/acceleration_test_list.cc#L330

**Review by jdduke — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@ANSHUMAN87 Can you please check @jdduke's comments and keep us posted. Thanks!

**Comment by ANSHUMAN87** [comment]
@gbaned : Thanks for tracking it! I have updated based on @jdduke comment for the internal failure reported by @rthadur .
Can you please approve it, and check for merge, as the new change is only to disable the test cases ? TIA!

**Review by rthadur — APPROVED** [review]

**Comment by ANSHUMAN87** [comment]
Thanks @jdduke, @rthadur, @gbaned!

**Pull request merged**